apache/polaris
# Enable eclipselink with H2 database in testing

# Description

The change is to support the testing against eclipselink with file-based H2 database in addition to in-memory implementation. That will test the env closer to the production. 
There is a difference between in-memory metastore and eclipseLink metastore: in-memory gets bootstrapped automatically since it's only used for test purpose while eclipseLink metastore doesn't boostrap automatically to avoid accidental wipeout in production.

To test out against  eclipseLink metastore, 
- The metastore gets bootstrapped for non in-memory metastore in tests (PolarisConnectionExtension).  
- The test DB location is placed in testDir and they will be cleaned up when the test starts to make the tests clean. 
- Remove persistence.xml in test and only keep one persistence.xml in main. I don't see value to keep two since the production version will always be a external version. 

TODO:

- There are some tests currently only testing against in-memory. We will need to add the coverage for elicplseLink.
- Instead of adding H2 dependencies to polaris-service, we can support loading the additional dependencies ondemand.


Fixes #149 

## Type of change

Please delete options that are not relevant.


# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] This has been tested against with the existing tests `./gradlew test` and all the tests passed against eclipselink.
- [x] This has been tested against with the existing tests `./gradlew test` and all the tests passed against in-memory.

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

Please delete options that are not relevant.

- [ ] If adding new functionality, I have discussed my implementation with the community using the linked GitHub issue
- [ ] I have signed and submitted the [ICLA](https://github.com/polaris-catalog/polaris/blob/main/ICLA.md) and if needed, the [CCLA](https://github.com/polaris-catalog/polaris/blob/main/CCLA.md). See [Contributing](https://github.com/polaris-catalog/polaris/blob/main/CONTRIBUTING.md) for details.
